ext/sqlite3/statement.c

```diff
@@ -121,6 +135,7 @@ step(VALUE self)
 
     TypedData_Get_Struct(self, sqlite3StmtRuby, &statement_type, ctx);
 
+    require_open_db(self);
```

Comment:
I didn't notice this in the initial review. `step` is an extremely hot method, do we actually need to do this?  I thought SQLite would already raise an exception if the db is somehow closed out from under it.

Reply:
Yeah, it does an IV read which is a bit much. If it was a struct member access I think it would be acceptable.

Reply:
I will look at other ways to make sure we're protected. This is really just protecting us from default if the DB was "discarded" so I can set a bit in the struct and that should be good enough.

Reply:
See https://github.com/sparklemotion/sqlite3-ruby/pull/565